src/sentry/tasks/commit_context.py

```diff
@@ -65,6 +82,11 @@ def queue_comment_task_if_needed(
         not pr.pullrequestcomment_set.exists()
         or group_owner.group_id not in pr.pullrequestcomment_set.get().group_ids
     ):
+        # create PR commit row for suspect commit and PR
+        pr_commit_query = PullRequestCommit.objects.filter(commit=commit, pull_request=pr)
```

Comment:
This is technically a race condition. Try Django ORM's `get_or_create`
https://docs.djangoproject.com/en/dev/ref/models/querysets/#get-or-create

Reply:
would this be addressed once i add the locks?

Reply:
Yeah, it should.